types.go

```diff
@@ -121,6 +121,15 @@ type functionInfo struct {
 	statuser         Statuser
 }
 
+func (f functionInfo) canEvaluate(grad, hess bool) error {
+	if grad && !(f.IsGradient || f.IsFunctionGradient) {
+		return fmt.Errorf("function does not implement Grad() nor FuncGrad()")
```

Comment:
Trending off topic. You could write, "does not implement Grad(), nor FuncGrad()", but it's beginning to sound dramatic, almost Shakspearean.